src/utils/helpers.rs

```diff
@@ -60,37 +60,17 @@ where
         }
     };
     let mut new_index = index as i32 + shift;
-    //Manually handle edge cases to allow more consistent behaviour
+    list.remove(index);
+    let v = &mut **list;
+
     if new_index < 0 {
```

Comment:
Isn't this basically the same statement as in #281?

Reply:
Not quite, as with the statement in cycle_vec each comparison is with 0. Whereas with the statement in reorder_vec is a comparison with two different numbers, 0 and len.

Reply:
oh, right. Probably should read more carefull befor jumping to conclusions :P